lib/CMakeLists.txt

```diff
@@ -0,0 +1,159 @@
+
+set(use_math_DEFS "")
+if(MSVC)
+  set(use_math_DEFS "-D_USE_MATH_DEFINES=1")
+endif()
+
+build_library_in_subdir(datetime)
+
+add_subdirectory(gis)
+
+build_library_in_subdir(driver
+  DEFS "${use_math_DEFS}"
+  DEPENDS grass_gis FREETYPE ICONV ZLIB)
+
+add_subdirectory(proj)
+
+file(GLOB raster_SRCS "./raster/*.c")
+list(APPEND raster_SRCS "./gis/gisinit.c")
+build_library_in_subdir(raster
+  DEFS "-DGDAL_DYNAMIC=1;-DGDAL_LINK=1"
+  SOURCES ${raster_SRCS}
+  DEPENDS GDAL PROJ grass_gproj)
+
+if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+  find_library(DL_LIBRARY dl)
+  mark_as_advanced(DL_LIBRARY)
+  target_link_libraries(grass_raster ${DL_LIBRARY})
+endif()
+
+configure_file(external/ccmath/ccmath.h
+  ${CMAKE_BINARY_DIR}/include/grass/ccmath_grass.h
+  COPYONLY)
+build_library_in_subdir(external/ccmath)
+
+build_library_in_subdir(external/shapelib NAME grass_shape
+  HEADERS "shapefil.h")
+
+build_library_in_subdir(gmath
+  DEPENDS grass_ccmath grass_gis
+  DEFS "${use_math_DEFS}"
+  OPTIONAL_DEPENDS FFTW)
+
+build_library_in_subdir(linkm)
+
+file(GLOB pngdriver_SRCS "./pngdriver/*.c")
+list(APPEND pngdriver_SRCS "./driver/init.c")
+build_library_in_subdir(pngdriver
+  SOURCES ${pngdriver_SRCS}
+  DEPENDS grass_driver LIBPNG ZLIB
+  INCLUDES "./driver")
+
+file(GLOB psdriver_SRCS "./psdriver/*.c")
+list(APPEND psdriver_SRCS "./driver/init.c")
+build_library_in_subdir(psdriver
+  SOURCES ${psdriver_SRCS}
+  DEPENDS grass_driver
+  INCLUDES "./driver")
+
+file(GLOB htmldriver_SRCS "./htmldriver/*.c")
+list(APPEND htmldriver_SRCS "./driver/init.c")
+build_library_in_subdir(htmldriver
+  SOURCES ${htmldriver_SRCS}
+  DEPENDS grass_driver
+  INCLUDES "./driver")
+
+set(_grass_display_DEFS)
+
+set(_grass_display_DEPENDS
+  grass_driver grass_raster
+  grass_htmldriver grass_pngdriver grass_psdriver)
+
+if(WITH_CAIRO)
+
+  set(_cairodriver_DEPENDS CAIRO FREETYPE grass_driver)
+  if(WITH_X11)
+    list(APPEND _cairodriver_DEPENDS X11)
+  endif()
+
+  file(GLOB cairodriver_SRCS "./cairodriver/*.c")
+  list(APPEND cairodriver_SRCS "./driver/init.c")
+  build_library_in_subdir(cairodriver
+    DEPENDS ${_cairodriver_DEPENDS}
+	SOURCES ${cairodriver_SRCS})
+
+  set(_grass_display_DEFS "-DUSE_CAIRO")
+  list(APPEND _grass_display_DEPENDS grass_cairodriver)
+endif()
+
+build_library_in_subdir(bitmap
+SOURCES "./bitmap/bitmap.c" "./bitmap/sparse.c"
+DEPENDS grass_linkm)
+
+build_library_in_subdir(btree)
+
+build_library_in_subdir(btree2 HEADERS "kdtree.h" DEPENDS grass_gis)
+
+build_library_in_subdir(display
+  DEFS ${_grass_display_DEFS}
+  DEPENDS ${_grass_display_DEPENDS})
+
+add_subdirectory(db)
```

Comment:
We should probably standardize "add_subdirectory" vs inlining the library defines. Else it will be displayed under different folders in Visual Studio. I prefer the former.

Reply:
I'm fine with your proposal. Please don't hesitate to create a PR on my branch.